test/trafficsplit/trafficsplit_test.go

```diff
@@ -0,0 +1,263 @@
+	}
+
+	if expected.rps != "" {
+		if expected.rps == "-" && actual.rps != "-" {
+			return fmt.Errorf("expected no rps for leaf %s, got '%s'", expected.leaf, actual.rps)
+		}
+
+		if expected.rps == zeroRPS && actual.rps != zeroRPS {
+			return fmt.Errorf("expected zero rps for leaf %s, got '%s'", expected.leaf, actual.rps)
+		}
+
+		if expected.rps != zeroRPS && actual.rps == zeroRPS {
+			return fmt.Errorf("expected non zero rps for leaf %s, got '%s'", expected.leaf, actual.rps)
+		}
+	}
+
+	if actual.latencyP50 != expected.latencyP50 && expected.latencyP50 != "" {
+		return fmt.Errorf("expected latencyP50 '%s' for leaf %s, got '%s'", expected.latencyP50, expected.leaf, actual.latencyP50)
+	}
+
+	if actual.latencyP95 != expected.latencyP95 && expected.latencyP95 != "" {
+		return fmt.Errorf("expected latencyP95 '%s' for leaf %s, got '%s'", expected.latencyP95, expected.leaf, actual.latencyP95)
+	}
+
+	if actual.latencyP99 != expected.latencyP99 && expected.latencyP99 != "" {
+		return fmt.Errorf("expected latencyP99 '%s' for leaf %s, got '%s'", expected.latencyP99, expected.leaf, actual.latencyP99)
+	}
+
+	return nil
+}
+
+func validateExpectedTsOutput(rows map[string]*statTsRow, expectedBackendSvc, expectedFailingSvc *statTsRow) error {
+	backendSvcLeafKey := "backend-svc"
+	backendFailingSvcLeafKey := "failing-svc"
+
+	backendRow, ok := rows[backendSvcLeafKey]
+	if !ok {
+		return fmt.Errorf("no stats found for [%s]", backendSvcLeafKey)
+	}
+
+	if err := validateTrafficSplit(backendRow, expectedBackendSvc); err != nil {
+		return err
+	}
+
+	backendFailingRow, ok := rows[backendFailingSvcLeafKey]
+	if !ok {
+		return fmt.Errorf("no stats found for [%s]", backendFailingSvcLeafKey)
+	}
+
+	if err := validateTrafficSplit(backendFailingRow, expectedFailingSvc); err != nil {
+		return err
+	}
+	return nil
+}
+
+func TestTrafficSplitCli(t *testing.T) {
+	out, _, err := TestHelper.LinkerdRun("inject", "--manual", "testdata/traffic_split_application.yaml")
+	if err != nil {
+		t.Fatalf("linkerd inject command failed\n%s", out)
+	}
+
+	prefixedNs := TestHelper.GetTestNamespace("trafficsplit-test")
+	err = TestHelper.CreateDataPlaneNamespaceIfNotExists(prefixedNs, nil)
+	if err != nil {
+		t.Fatalf("failed to create %s namespace: %s", prefixedNs, err)
+	}
+	out, err = TestHelper.KubectlApply(out, prefixedNs)
+	if err != nil {
+		t.Fatalf("kubectl apply command failed\n%s", out)
+	}
+
+	// wait for deployments to start
+	for _, deploy := range []string{"backend", "failing", "slow-cooker"} {
+		if err := TestHelper.CheckPods(prefixedNs, deploy, 1); err != nil {
+			t.Error(err)
+		}
+
+		if err := TestHelper.CheckDeployment(prefixedNs, deploy, 1); err != nil {
+			t.Error(fmt.Errorf("Error validating deployment [%s]:\n%s", deploy, err))
+		}
+	}
+
+	t.Run("ensure traffic is sent to one backend only", func(t *testing.T) {
+		err := TestHelper.RetryFor(40*time.Second, func() error {
```

Comment:
Food for thought, are the sleep and the retryFor redundant?  Is it possible to just use the retryFor to keep trying until the old stats fall out of the window?